### PR TITLE
W-12042900: Add AST Validation to forbid components not belonging to any Extension Model

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java [hunk 1]
```diff
@@ -47,6 +47,7 @@ public List<Validation> get() {
                                                           new NamedTopLevelElementsHaveName(),
                                                           new NameHasValidCharacters(),
                                                           new NameIsNotRepeated(),
+                                                          new AllComponentsBelongToSomeExtensionModel(),
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java:50]: maybe this should run earlier, beofre other validations that rely on the data from the extension model?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java (hunk 1 above)
Comment (eze210) [reply runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java:50]: makes sense

Comment (eze210) [reply runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java]: done

**Review by elrodro83 — DISMISSED** [review]

**Comment by eze210** [comment]
--validate

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/context/DynamicServerNotificationManagerConfiguratorTestCase.java [hunk 2]
```diff
@@ -29,8 +30,8 @@ public class DynamicServerNotificationManagerConfiguratorTestCase extends Abstra
   public SystemProperty systemProperty = new SystemProperty(DISABLE_REGISTRY_BOOTSTRAP_OPTIONAL_ENTRIES_PROPERTY, "false");
 
   @Override
-  protected ConfigurationBuilder getBuilder() throws Exception {
-    return new SpringXmlConfigurationBuilder(new String[] {"./dynamic-server-notification-manager-test.xml"}, emptyMap());
```
Comment (elrodro83) [inline modules/spring-config/src/test/java/org/mule/runtime/config/internal/context/DynamicServerNotificationManagerConfiguratorTestCase.java:33]: why is this removed?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/spring-config/src/test/java/org/mule/runtime/config/internal/context/DynamicServerNotificationManagerConfiguratorTestCase.java (hunk 2 above)
Comment (eze210) [reply modules/spring-config/src/test/java/org/mule/runtime/config/internal/context/DynamicServerNotificationManagerConfiguratorTestCase.java:33]: mistake

**Review by elrodro83 — APPROVED** [review]
